src/main/scala/sigmastate/eval/TreeBuilding.scala

```diff
@@ -48,7 +48,7 @@ trait TreeBuilding extends RuntimeCosting { IR: IRContext =>
   type DefEnv = Map[Sym, (Int, SType)]
 
   object IsArithOp {
-    def unapply(op: EndoBinOp[_]): Option[Byte] = op match {
+    def unapply(op: EndoBinOp[_]): Option[OpCode] = op match {
```

Comment:
This code is part of transaction validation. So performance is critical. 
This however will produce a lot of Some object in the heap.
Nullable will not help here as well.
So please reimplement avoiding allocations on a code pass.

Reply:
Done.